### ISSUE TITLE
Mutex for fibers 

### ISSUE THREAD

**Issue by publicocean0** [dwango/fibers-rs#23]
How do you handle a mutex with fibers? There is a library for fibers mutex?

**Comment by koba-e964** [comment]
Do you want something like [`futures_locks::Mutex`](https://docs.rs/futures-locks/0.3.1/futures_locks/struct.Mutex.html), but tailored for `fibers-rs`?
Then unfortunately I don't think that there's a library for this particular purpose. I don't think `futures_locks::Mutex` works either because `fibers-rs` uses its own mechanism for task scheduling.

**Comment by publicocean0** [comment]
yes the problem is if there is a lock in normal version , the thread is suspended until signal notification instead in fibers scheduler it might be  swiched to another fiber. The perfomance is higher  . So i have to realize a custom lock library pratically

**Comment by publicocean0** [comment]
sorry a question : but pratically also futures in rust are using continuation concept , right? is a different way to implement the same thing?
